### ISSUE TITLE
multi_class：训练时 train.py: error: argument --load_best_model_at_end: expected one argument，怎么解决？

### ISSUE THREAD

**Issue by done520** [PaddlePaddle/PaddleNLP#4929]
### 请提出你的问题

使用multi_class例子进行文本分类训练（https://github.com/PaddlePaddle/PaddleNLP/tree/develop/applications/text_classification/multi_class#readme），训练完全follow README，代码是直接git pull 下来的，训练提示--load_best_model_at_end错误，命令：
$ python train.py \
>     --do_train \
>     --do_eval \
>     --do_export \
>     --model_name_or_path ernie-3.0-tiny-medium-v2-zh \
>     --output_dir checkpoint \
>     --device gpu \
>     --num_train_epochs 100 \
>     --early_stopping True \
>     --early_stopping_patience 5 \
>     --learning_rate 3e-5 \
>     --max_length 128 \
>     --per_device_eval_batch_size 32 \
>     --per_device_train_batch_size 32 \
>     --metric_for_best_model accuracy \
>     --load_best_model_at_end \
>     --logging_steps 5 \
>     --evaluation_strategy epoch \
>     --save_strategy epoch \
>     --save_total_limit 1

包：
paddlenlp                    2.5.1
paddlepaddle-gpu             2.4.0rc0.post101

提示：
Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
  declare_namespace(pkg)
/home/user/.local/lib/python3.7/site-packages/pkg_resources/__init__.py:2804: DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('mpl_toolkits')`.
Implementing implicit namespace packages (as specified in PEP 420) is preferred to `pkg_resources.declare_namespace`. See https://setuptools.pypa.io/en/latest/references/keywords.html#keyword-namespace-packages
  declare_namespace(pkg)
/home/user/.local/lib/python3.7/site-packages/_distutils_hack/__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")
usage: train.py [-h] [--model_name_or_path MODEL_NAME_OR_PATH]
                [--export_model_dir EXPORT_MODEL_DIR]
                [--max_length MAX_LENGTH] [--early_stopping [EARLY_STOPPING]]
                [--early_stopping_patience EARLY_STOPPING_PATIENCE]
                [--debug [DEBUG]] [--train_path TRAIN_PATH]
                [--dev_path DEV_PATH] [--test_path TEST_PATH]
                [--label_path LABEL_PATH] [--bad_case_path BAD_CASE_PATH]
                --output_dir OUTPUT_DIR
                [--overwrite_output_dir [OVERWRITE_OUTPUT_DIR]]
                [--do_train [DO_TRAIN]] [--do_eval [DO_EVAL]]
                [--do_predict [DO_PREDICT]] [--do_export [DO_EXPORT]]
                [--evaluation_strategy {no,steps,epoch}]
                [--prediction_loss_only [PREDICTION_LOSS_ONLY]]
                [--per_device_train_batch_size PER_DEVICE_TRAIN_BATCH_SIZE]
                [--per_device_eval_batch_size PER_DEVICE_EVAL_BATCH_SIZE]
                [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS]
                [--learning_rate LEARNING_RATE] [--weight_decay WEIGHT_DECAY]
                [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2]
                [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM]
                [--num_train_epochs NUM_TRAIN_EPOCHS] [--max_steps MAX_STEPS]
                [--lr_scheduler_type LR_SCHEDULER_TYPE]
                [--warmup_ratio WARMUP_RATIO] [--warmup_steps WARMUP_STEPS]
                [--log_on_each_node [LOG_ON_EACH_NODE]]
                [--no_log_on_each_node] [--logging_dir LOGGING_DIR]
                [--logging_strategy {no,steps,epoch}]
                [--logging_first_step [LOGGING_FIRST_STEP]]
                [--logging_steps LOGGING_STEPS]
                [--save_strategy {no,steps,epoch}] [--save_steps SAVE_STEPS]
                [--save_total_limit SAVE_TOTAL_LIMIT]
                [--save_on_each_node [SAVE_ON_EACH_NODE]]
                [--no_cuda [NO_CUDA]] [--seed SEED] [--bf16 [BF16]]
                [--fp16 [FP16]] [--fp16_opt_level FP16_OPT_LEVEL]
                [--bf16_full_eval [BF16_FULL_EVAL]]
                [--fp16_full_eval [FP16_FULL_EVAL]] [--sharding SHARDING]
                [--sharding_degree SHARDING_DEGREE] [--recompute [RECOMPUTE]]
                [--scale_loss SCALE_LOSS]
                [--minimum_eval_times MINIMUM_EVAL_TIMES]
                [--local_rank LOCAL_RANK]
                [--dataloader_drop_last [DATALOADER_DROP_LAST]]
                [--eval_steps EVAL_STEPS]
                [--dataloader_num_workers DATALOADER_NUM_WORKERS]
                [--past_index PAST_INDEX] [--run_name RUN_NAME]
                [--device DEVICE] [--disable_tqdm DISABLE_TQDM]
                [--remove_unused_columns REMOVE_UNUSED_COLUMNS]
                [--label_names LABEL_NAMES [LABEL_NAMES ...]]
                [--load_best_model_at_end LOAD_BEST_MODEL_AT_END]
                [--metric_for_best_model METRIC_FOR_BEST_MODEL]
                [--greater_is_better GREATER_IS_BETTER]
                [--ignore_data_skip [IGNORE_DATA_SKIP]] [--optim OPTIM]
                [--report_to REPORT_TO [REPORT_TO ...]]
                [--resume_from_checkpoint RESUME_FROM_CHECKPOINT]
                [--skip_memory_metrics [SKIP_MEMORY_METRICS]]
                [--no_skip_memory_metrics]
                [--flatten_param_grads FLATTEN_PARAM_GRADS]
                [--lazy_data_processing LAZY_DATA_PROCESSING]
                [--do_compress [DO_COMPRESS]] [--input_dtype INPUT_DTYPE]
                [--prune_embeddings [PRUNE_EMBEDDINGS]]
                [--onnx_format ONNX_FORMAT] [--strategy STRATEGY]
                [--width_mult_list WIDTH_MULT_LIST [WIDTH_MULT_LIST ...]]
                [--weight_quantize_type WEIGHT_QUANTIZE_TYPE]
                [--activation_quantize_type ACTIVATION_QUANTIZE_TYPE]
                [--algo_list ALGO_LIST [ALGO_LIST ...]]
                [--batch_num_list BATCH_NUM_LIST [BATCH_NUM_LIST ...]]
                [--batch_size_list BATCH_SIZE_LIST [BATCH_SIZE_LIST ...]]
                [--round_type ROUND_TYPE] [--bias_correction BIAS_CORRECTION]
                [--input_infer_model_path INPUT_INFER_MODEL_PATH]
                [--use_pact USE_PACT] [--moving_rate MOVING_RATE]
train.py: error: argument --load_best_model_at_end: expected one argument

**Comment by LemonNoel** [comment]
需要增加参数值 `--load_best_model_at_end True`，后续我们修改一下文档。

**Comment by done520** [comment]
应该还其他地方有问题，train.py: error: argument --load_best_model_at_end: invalid typing.Union[bool, NoneType] value: 'True'

**Comment by done520** [comment]
@LemonNoel 想问一下，如果是长文本，大于512的做分类有什么建议吗？截断后做overlap吗？

**Comment by lugimzzz** [comment]
我们这块没有复现出这个问题，你可以`pip list`确定一下版本是2.5.1，并保证使用的是脚本提供的命令行。

**Comment by lugimzzz** [comment]
> @LemonNoel 想问一下，如果是长文本，大于512的做分类有什么建议吗？截断后做overlap吗？

目前ernie 3.0 输入最大支持2048，考虑到显存和效果问题，可以使用截断、滑窗截断、根据数据特点对长文本进行预处理

**Comment by done520** [comment]
@LemonNoel 感谢。我确认我这边就是paddlenlp  为2.5.1，paddle为2.4                  
1. 依赖包：
paddle-bfloat                0.1.7
paddle2onnx                  1.0.5
paddlefsl                    1.1.0
paddlenlp                    2.5.1
paddlepaddle-gpu             2.4.0rc0.post101

![image](https://user-images.githubusercontent.com/111326875/220541485-3493efed-1c62-4528-adcb-3bbdfc763ced.png)

2. 运行命令：
我是自己pull 整个项目到服务器，然后在cd 至 multi_class目录下运行的，数据也是KUAKE_QIC
python train.py \
    --do_train \
    --do_eval \
    --do_export \
    --model_name_or_path ernie-3.0-tiny-medium-v2-zh \
    --output_dir checkpoint \
    --device gpu \
    --num_train_epochs 100 \
    --early_stopping True \
    --early_stopping_patience 5 \
    --learning_rate 3e-5 \
    --max_length 128 \
    --per_device_eval_batch_size 32 \
    --per_device_train_batch_size 32 \
    --metric_for_best_model accuracy \
    --load_best_model_at_end \
    --logging_steps 5 \
    --evaluation_strategy epoch \
    --save_strategy epoch \
    --save_total_limit 1

**Comment by done520** [comment]
我卸载paddlenlp后重新安装 pip install paddlenlp,

重新运行，命令：python train.py   --do_train   --do_eval   --do_export   --model_name_or_path ernie-3.0-tiny-medium-v2-zh  --output_dir checkpoint --device cpu --num_train_epochs 100 --early_stopping True --early_stopping_patience 5 --learning_rate 3e-5  --max_length 128 --per_device_eval_batch_size 32 --per_device_train_batch_size 32 --metric_for_best_model accuracy  --load_best_model_at_end True --logging_steps 5 --evaluation_strategy epoch  --save_strategy epoch --save_total_limit 1

出现这么个问题：
usage: train.py [-h] [--model_name_or_path MODEL_NAME_OR_PATH]
                [--data_dir DATA_DIR] [--max_seq_length MAX_SEQ_LENGTH]
                [--early_stopping_patience EARLY_STOPPING_PATIENCE]
                --output_dir OUTPUT_DIR
                [--overwrite_output_dir [OVERWRITE_OUTPUT_DIR]]
                [--do_train [DO_TRAIN]] [--do_eval [DO_EVAL]]
                [--do_predict [DO_PREDICT]] [--do_export [DO_EXPORT]]
                [--evaluation_strategy {no,steps,epoch}]
                [--prediction_loss_only [PREDICTION_LOSS_ONLY]]
                [--per_device_train_batch_size PER_DEVICE_TRAIN_BATCH_SIZE]
                [--per_device_eval_batch_size PER_DEVICE_EVAL_BATCH_SIZE]
                [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS]
                [--learning_rate LEARNING_RATE] [--weight_decay WEIGHT_DECAY]
                [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2]
                [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM]
                [--num_train_epochs NUM_TRAIN_EPOCHS] [--max_steps MAX_STEPS]
                [--lr_scheduler_type LR_SCHEDULER_TYPE]
                [--warmup_ratio WARMUP_RATIO] [--warmup_steps WARMUP_STEPS]
                [--log_on_each_node [LOG_ON_EACH_NODE]]
                [--no_log_on_each_node] [--logging_dir LOGGING_DIR]
                [--logging_strategy {no,steps,epoch}]
                [--logging_first_step [LOGGING_FIRST_STEP]]
                [--logging_steps LOGGING_STEPS]
                [--save_strategy {no,steps,epoch}] [--save_steps SAVE_STEPS]
                [--save_total_limit SAVE_TOTAL_LIMIT]
                [--save_on_each_node [SAVE_ON_EACH_NODE]]
                [--no_cuda [NO_CUDA]] [--seed SEED] [--bf16 [BF16]]
                [--fp16 [FP16]] [--fp16_opt_level FP16_OPT_LEVEL]
                [--bf16_full_eval [BF16_FULL_EVAL]]
                [--fp16_full_eval [FP16_FULL_EVAL]] [--sharding SHARDING]
                [--sharding_degree SHARDING_DEGREE] [--recompute [RECOMPUTE]]
                [--scale_loss SCALE_LOSS]
                [--minimum_eval_times MINIMUM_EVAL_TIMES]
                [--local_rank LOCAL_RANK]
                [--dataloader_drop_last [DATALOADER_DROP_LAST]]
                [--eval_steps EVAL_STEPS]
                [--dataloader_num_workers DATALOADER_NUM_WORKERS]
                [--past_index PAST_INDEX] [--run_name RUN_NAME]
                [--device DEVICE] [--disable_tqdm DISABLE_TQDM]
                [--remove_unused_columns REMOVE_UNUSED_COLUMNS]
                [--label_names LABEL_NAMES [LABEL_NAMES ...]]
                [--load_best_model_at_end LOAD_BEST_MODEL_AT_END]
                [--metric_for_best_model METRIC_FOR_BEST_MODEL]
                [--greater_is_better GREATER_IS_BETTER]
                [--ignore_data_skip [IGNORE_DATA_SKIP]] [--optim OPTIM]
                [--report_to REPORT_TO [REPORT_TO ...]]
                [--resume_from_checkpoint RESUME_FROM_CHECKPOINT]
                [--skip_memory_metrics [SKIP_MEMORY_METRICS]]
                [--no_skip_memory_metrics]
                [--flatten_param_grads FLATTEN_PARAM_GRADS]
                [--lazy_data_processing LAZY_DATA_PROCESSING]
train.py: error: argument --early_stopping_patience: invalid int value: 'True'

**Comment by lugimzzz** [comment]
这块看上去你的argument这块有问题，请问你是怎么运行脚本的，有什么特殊方式吗？
你试一下创建一个args.py 包含一下内容，然后运行一下`python args.py --early_stopping_patience 5`看看有没有问题
```
from paddlenlp.trainer import PdArgumentParser
from dataclasses import dataclass, field
@dataclass
class DataArguments:
    early_stopping_patience: int = field(default=4, metadata={"help": "Stop training when the specified metric worsens for early_stopping_patience evaluation calls"})

parser = PdArgumentParser((DataArguments))
data_args = parser.parse_args_into_dataclasses()[0]
print(data_args.early_stopping_patience)
```

**Comment by WilliamsLiang** [comment]
环境安装了modelscope后，出现invalid typing.Union[bool, NoneType] value
![image](https://user-images.githubusercontent.com/30794803/232183039-295e8c36-a24e-41f3-9e6f-52f897f9de55.png)
怀疑是某个包版本冲突引起的，但是无法定位

**Comment by ttgopen** [comment]
![0001](https://user-images.githubusercontent.com/122525537/232680553-15ed65dc-daaf-4cc6-970c-38758f8b62b3.png)

**Comment by lugimzzz** [comment]
详细描述一下问题？

**Comment by lugimzzz** [comment]
如果命令行不方便传参可以用这种方式
args_list = ['--output_dir', 'checkpoint']
model_args, data_args, training_args = parser.parse_args_into_dataclasses(args=args_list)
